m4/ax_lua.m4

```diff
@@ -0,0 +1,664 @@
+#
+#   If headers are found, then ACTION-IF-FOUND is performed, otherwise
+#   ACTION-IF-NOT-FOUND is performed. If ACTION-IF-NOT-FOUND is blank, then
+#   it will default to printing an error. To prevent the default behavior,
+#   set the action to ':'.
+#
+#   AX_LUA_LIBS: Search for Lua libraries. Requires that AX_PROG_LUA be
+#   expanded before this macro. Adds precious variable LUA_LIB, which may
+#   contain Lua specific linker flags, e.g. -llua5.1. If LUA_LIB is blank,
+#   then this macro will attempt to find suitable flags.
+#
+#   LUA_LIB can be used by Automake to link Lua modules or executables with
+#   embedded interpreters. The *_LIBADD and *_LDADD variables should be used
+#   for this purpose, e.g. mymod_LIBADD = $(LUA_LIB).
+#
+#   This macro searches for the Lua library. More technically, it searches
+#   for a library containing the function lua_load. The search is performed
+#   with a combination of LIBS, LIBRARY_PATH, and LUA_LIB.
+#
+#   If the search determines that some linker flags are missing, then those
+#   flags will be added to LUA_LIB.
+#
+#   If libraries are found, then ACTION-IF-FOUND is performed, otherwise
+#   ACTION-IF-NOT-FOUND is performed. If ACTION-IF-NOT-FOUND is blank, then
+#   it will default to printing an error. To prevent the default behavior,
+#   set the action to ':'.
+#
+#   AX_LUA_READLINE: Search for readline headers and libraries. Requires the
+#   AX_LIB_READLINE macro, which is provided by ax_lib_readline.m4 from the
+#   Autoconf Archive.
+#
+#   If a readline compatible library is found, then ACTION-IF-FOUND is
+#   performed, otherwise ACTION-IF-NOT-FOUND is performed.
+#
+# LICENSE
+#
+#   Copyright (c) 2015 Reuben Thomas <rrt@sc3d.org>
+#   Copyright (c) 2014 Tim Perkins <tprk77@gmail.com>
+#
+#   This program is free software: you can redistribute it and/or modify it
+#   under the terms of the GNU General Public License as published by the
+#   Free Software Foundation, either version 3 of the License, or (at your
+#   option) any later version.
+#
+#   This program is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+#   Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License along
+#   with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+#   As a special exception, the respective Autoconf Macro's copyright owner
+#   gives unlimited permission to copy, distribute and modify the configure
+#   scripts that are the output of Autoconf when processing the Macro. You
+#   need not follow the terms of the GNU General Public License when using
+#   or distributing such scripts, even though portions of the text of the
+#   Macro appear in them. The GNU General Public License (GPL) does govern
+#   all other use of the material that constitutes the Autoconf Macro.
```

Comment:
This file cannot be included in the project since it uses the GPL and this exception does not apply when including the file directly

Reply:
@jacobkeeler considering licensing issues I propose to create cmake based build system for bson4c lib. 
But I guess that we probably can miss terms for including #3473 in 7.0 release.
Form my perspective is not huge issue, because its infrastructure bug but not an SDL bug and should not affect SDL business logic on hardware.

Reply:
@jacobkeeler Please advise if you agree with the approach mentioned by @LuxoftAKutsan: 

> I propose to create cmake based build system for bson4c lib.

Reply:
@LuxoftAKutsan @jacobkeeler Let me put some comments on this issue.
I agree that the issue is not related to `sdl_core`. 
However it has a huge impact on current CI infrastructure. Besides the fact we have failed builds on our internal CI, we also have the same issues at our Open CI. E.g.: https://opensdl-jenkins.prjdmz.luxoft.com/view/PR_checks/job/OpenSDL_PR/2623/console
Having this issue we can not guarantee full automatic process of SDL builds.
On every fail developer has to check manually the reason of that and in case of the issue restart the build.
It's time consuming and increase load on CI.

So I would appreciate if we able to come up with the solution.

Reply:
@dboltovskyi @LuxoftAKutsan after looking into this a bit further, I was able to find a potential solution, let me know if it works on your end: https://github.com/smartdevicelink/bson_c_lib/pull/52

Reply:
@jacobkeeler Thank you for your update regarding this issue.
We will check your solution at our end and get back with results.

Reply:
@jacobkeeler It looks like your solution works on our end. We haven't received build failures in ~100 runs in a row having different SDL configurations.